http/logical.go

```diff
@@ -199,7 +199,7 @@ func buildLogicalRequestNoAuth(perfStandby bool, w http.ResponseWriter, r *http.
 		req.HTTPRequest = r
 	}
 	if responseWriter != nil {
-		req.ResponseWriter = logical.NewHTTPResponseWriter(responseWriter)
+		req.ResponseWriter = logical.NewStatusHeaderResponseWriter(responseWriter)
```

Comment:
Why do we need to re-wrap?  Why can't we just assign responseWriter directly?

Reply:
So, responseWriter is of type http.ResponseWriter. if I assign that directly, then, where ever we would need extra functionality like `ResponseWriter.Written()`, we would need to assert the `WrappingResponseWriter` interface first. Although there are not many places right now for such functionality, it is easy to miss fixing those. It is not just the `Written()` function as well.

Reply:
Is it though?  I thought everything was wrapped by wrapGenericHandler, and it's always passing down a response writer of the new type.

Reply:
hmm, I think so? The function signature of this function defines `w http.ResponseWriter` which is assigned to `responseWriter`. But, another thing is that of which type should the `req.ResponseWriter` be defined? should it be `http.ResponseWriter`, or `logical.StatusHeaderResponseWriter`?

Reply:
I wasn't speaking in terms of function signature, I'm saying that since wrapGenericHandler is wrapping all request handlers, we shouldn't need to wrap the response handler again.  And if we don't need to handle re-wrapping I think some of the code will be simplified.

I think we should probably stick with http.ResponseWriter, but that's mostly because I expect trying to switch it would cause problems.  If it's possible to use StatusHeaderResponseWriter without it being really disruptive, that could make sense.

Reply:
Sounds good! I will stick with http.ResponseWriter.